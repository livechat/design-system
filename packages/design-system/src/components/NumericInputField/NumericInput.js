import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './style.scss';
import getMergedClassNames from '../../utils/getMergedClassNames';
import { KeyCodes } from '../../constants/keyCodes';

const cx = classNames.bind(styles);
const baseClass = 'numeric-input';

class NumericInput extends React.PureComponent {
  componentDidMount() {
    if (this.props.max && parseInt(this.props.value, 10) > this.props.max) {
      this.callOnChange(this.props.max);
    }

    if (this.props.min && parseInt(this.props.value, 10) < this.props.min) {
      this.callOnChange(this.props.min);
    }
  }

  componentWillUnmount() {
    this.removeKeyboardEventListeners();
  }

  onKeyDown = e => {
    if (e.keyCode === KeyCodes.arrowDown) {
      e.preventDefault();
      this.changeValue(-1);
    }
    if (e.keyCode === KeyCodes.arrowUp) {
      e.preventDefault();
      this.changeValue(1);
    }
  };

  getComponentStyles = () => {
    const componentStyles = {
      ...(this.props.style || {})
    };
    if (this.props.width) {
      componentStyles.width = this.props.width;
    }

    return componentStyles;
  };

  addKeyboardEventListeners = () => {
    document.addEventListener('keydown', this.onKeyDown);
  };

  removeKeyboardEventListeners = () => {
    document.removeEventListener('keydown', this.onKeyDown);
  };

  changeValue = val => {
    if (this.props.value !== '' && this.props.value !== '-') {
      this.callOnChange(this.calcValue(parseInt(this.props.value, 10) + val));
    } else if (this.props.min && val < this.props.min) {
      this.callOnChange(this.props.min);
    } else if (this.props.max && val > this.props.max) {
      this.callOnChange(this.props.max);
    } else {
      this.callOnChange(val);
    }
  };

  handleChange = e => {
    e.preventDefault();
    e.stopPropagation();
    const { value } = this.props;
    const inputVal = e.target.value.replace(/((?!([-]|([-]?\d+))).)/, '');

    if (inputVal === '') {
      this.callOnChange('');
      return;
    }

    if (inputVal === '-') {
      this.callOnChange('-');
      return;
    }

    const val = parseInt(inputVal, 10);

    if (String(val) !== inputVal) {
      this.callOnChange(value);
    } else {
      const calculatedValue = this.calcValue(val);
      this.callOnChange(calculatedValue);
    }
  };

  callOnChange = value => this.props.onChange(String(value));

  calcValue = val => {
    const { max, min } = this.props;

    if (max && val > max) {
      return max;
    }

    if (min && val < min) {
      return min;
    }
    return val;
  };

  handleIncrementClick = () => {
    this.changeValue(1);
  };

  handleDecrementClick = () => {
    this.changeValue(-1);
  };

  handleFocus = () => {
    this.addKeyboardEventListeners();
  };

  handleBlur = () => {
    this.removeKeyboardEventListeners();
  };

  render() {
    const {
      error,
      className,
      max,
      min,
      onChange,
      noControls,
      value,
      disabled,
      style,
      ...restProps
    } = this.props;

    const mergedClassNames = getMergedClassNames(
      cx({
        [baseClass]: true,
        [`${baseClass}--error`]: error,
        [`${baseClass}--no-controls`]: noControls,
        [`${baseClass}--disabled`]: disabled
      }),
      className
    );

    return (
      <div className={mergedClassNames} style={this.getComponentStyles()}>
        <input
          type="text"
          {...restProps}
          value={value}
          disabled={disabled}
          onChange={this.handleChange}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
        {!noControls && (
          <React.Fragment>
            <button
              tabIndex="-1"
              disabled={
                disabled ||
                (this.props.max &&
                  parseInt(this.props.value, 10) === this.props.max)
              }
              onClick={this.handleIncrementClick}
              aria-label="Increment value"
              className={styles[`${baseClass}__increment`]}
              type="button"
            />
            <button
              tabIndex="-1"
              disabled={
                disabled ||
                (this.props.min &&
                  parseInt(this.props.value, 10) === this.props.min)
              }
              aria-label="Decrement value"
              className={styles[`${baseClass}__decrement`]}
              onClick={this.handleDecrementClick}
              type="button"
            />
          </React.Fragment>
        )}
      </div>
    );
  }
}

NumericInput.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  error: PropTypes.string,
  value: PropTypes.string,
  max: PropTypes.number,
  min: PropTypes.number,
  disabled: PropTypes.bool,
  noControls: PropTypes.bool,
  width: PropTypes.string,
  /**
   * It's called with string value: '', '-' or '{number}'
   */
  onChange: PropTypes.func.isRequired
};

export default NumericInput;
