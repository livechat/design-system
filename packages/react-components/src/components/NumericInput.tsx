import * as React from 'react';
import cx from 'classnames';
import {
  ChevronUp,
  ChevronDown,
} from '@livechat/design-system-icons/dist/material';
import { Icon } from './Icon';

enum EventKeys {
  ArrowUp = 'ArrowUp',
  ArrowDown = 'ArrowDown',
}

const baseClass = 'lc-numeric-input';

export interface IProps {
  className?: string;
  error?: string;
  value: string;
  max?: number;
  min?: number;
  disabled?: boolean;
  noControls?: boolean;
  onChange: (value: string) => void;
}

export type INumericInputProps = IProps &
  React.InputHTMLAttributes<HTMLInputElement>;

export const NumericInput: React.FC<INumericInputProps> = ({
  className,
  error,
  value,
  max,
  min,
  disabled,
  noControls,
  style,
  onChange,
  ...restProps
}) => {
  const mergedClassNames = cx(
    baseClass,
    {
      [`${baseClass}--error`]: error,
      [`${baseClass}--no-controls`]: noControls,
      [`${baseClass}--disabled`]: disabled,
    },
    className
  );

  const handleOnFocus = () =>
    window.addEventListener('keydown', onKeyDown, true);

  const handleOnBlur = () => window.removeEventListener('keydown', onKeyDown);

  const callOnChange = (val: number | string) => onChange(String(val));

  const calcValue = (val: number) => {
    if (max !== undefined && val > max) {
      return max;
    }

    if (min !== undefined && val < min) {
      return min;
    }

    return val;
  };

  const updateValue = (val: number) => {
    const newValue = parseInt(value, 10) + val;

    return callOnChange(calcValue(newValue));
  };

  const onKeyDown = (e: KeyboardEvent) => {
    if (e.key === EventKeys.ArrowDown) {
      e.preventDefault();
      updateValue(-1);
    }

    if (e.key === EventKeys.ArrowUp) {
      e.preventDefault();
      updateValue(1);
    }
  };

  const hasReachedTheLimit = (value: string, margin?: number) =>
    margin !== undefined && parseInt(value, 10) === margin;

  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
    e.preventDefault();
    e.stopPropagation();
    const inputVal = e.currentTarget.value.replace(
      /((?!([-]|([-]?\d+))).)/,
      ''
    );

    if (inputVal === '' || inputVal === '-') {
      return callOnChange(inputVal);
    }

    const newValue = parseInt(inputVal, 10);

    return callOnChange(calcValue(newValue));
  };

  const handleIncrementClick = () => {
    return updateValue(1);
  };

  const handleDecrementClick = () => {
    return updateValue(-1);
  };

  return (
    <div className={mergedClassNames} style={style}>
      <input
        type="text"
        {...restProps}
        value={value}
        disabled={disabled}
        onChange={handleOnChange}
        onFocus={handleOnFocus}
        onBlur={handleOnBlur}
        min={min}
        max={max}
      />
      {!noControls && (
        <>
          <button
            tabIndex={-1}
            disabled={disabled || hasReachedTheLimit(value, max)}
            onClick={handleIncrementClick}
            aria-label="Increment value"
            className={`${baseClass}__increment`}
            type="button"
          >
            <Icon
              source={ChevronUp}
              disabled={disabled || hasReachedTheLimit(value, max)}
            />
          </button>
          <button
            tabIndex={-1}
            disabled={disabled || hasReachedTheLimit(value, min)}
            aria-label="Decrement value"
            className={`${baseClass}__decrement`}
            onClick={handleDecrementClick}
            type="button"
          >
            <Icon
              source={ChevronDown}
              disabled={disabled || hasReachedTheLimit(value, min)}
            />
          </button>
        </>
      )}
    </div>
  );
};
