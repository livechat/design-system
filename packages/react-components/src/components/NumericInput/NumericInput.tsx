import * as React from 'react';
import cx from 'clsx';
import {
  ChevronUp,
  ChevronDown,
} from '@livechat/design-system-icons/react/material';
import { Icon, IconTypeName } from '../Icon';
import { KeyCodes } from '../../utils/keyCodes';

import styles from './NumericInput.module.scss';

const baseClass = 'numeric-input';

interface Props {
  className?: string;
  error?: string;
  value: string;
  max?: number;
  min?: number;
  disabled?: boolean;
  noControls?: boolean;
  onChange: (value: string) => void;
}

export type NumericInputProps = Props &
  React.InputHTMLAttributes<HTMLInputElement>;

export const NumericInput: React.FC<NumericInputProps> = ({
  className,
  error,
  value,
  max,
  min,
  disabled,
  noControls,
  style,
  onChange,
  ...restProps
}) => {
  const mergedClassNames = cx(
    styles[baseClass],
    {
      [styles[`${baseClass}--error`]]: error,
      [styles[`${baseClass}--no-controls`]]: noControls,
      [styles[`${baseClass}--disabled`]]: disabled,
    },
    className
  );

  const handleOnFocus = () =>
    window.addEventListener('keydown', onKeyDown, true);

  const handleOnBlur = () => window.removeEventListener('keydown', onKeyDown);

  const callOnChange = (val: number | string) => onChange(String(val));

  const calcValue = (val: number) => {
    if (max !== undefined && val > max) {
      return max;
    }

    if (min !== undefined && val < min) {
      return min;
    }

    return val;
  };

  const updateValue = (val: number) => {
    const newValue = parseInt(value, 10) + val;

    return callOnChange(calcValue(newValue));
  };

  const onKeyDown = (e: KeyboardEvent) => {
    if (e.key === KeyCodes.arrowDown) {
      e.preventDefault();
      updateValue(-1);
    }

    if (e.key === KeyCodes.arrowUp) {
      e.preventDefault();
      updateValue(1);
    }
  };

  const hasReachedTheLimit = (value: string, margin?: number) =>
    margin !== undefined && parseInt(value, 10) === margin;

  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
    e.preventDefault();
    e.stopPropagation();
    const inputVal = e.currentTarget.value.replace(
      /((?!([-]|([-]?\d+))).)/,
      ''
    );

    if (inputVal === '' || inputVal === '-') {
      return callOnChange(inputVal);
    }

    const newValue = parseInt(inputVal, 10);

    return callOnChange(calcValue(newValue));
  };

  const handleIncrementClick = () => {
    return updateValue(1);
  };

  const handleDecrementClick = () => {
    return updateValue(-1);
  };

  return (
    <div className={mergedClassNames} style={style}>
      <input
        type="text"
        {...restProps}
        value={value}
        disabled={disabled}
        onChange={handleOnChange}
        onFocus={handleOnFocus}
        onBlur={handleOnBlur}
        min={min}
        max={max}
      />
      {!noControls && (
        <>
          <button
            tabIndex={-1}
            disabled={disabled || hasReachedTheLimit(value, max)}
            onClick={handleIncrementClick}
            aria-label="Increment value"
            className={styles[`${baseClass}__increment`]}
            type="button"
          >
            <Icon
              source={ChevronUp}
              disabled={disabled || hasReachedTheLimit(value, max)}
              iconType={IconTypeName.Primary}
            />
          </button>
          <button
            tabIndex={-1}
            disabled={disabled || hasReachedTheLimit(value, min)}
            aria-label="Decrement value"
            className={styles[`${baseClass}__decrement`]}
            onClick={handleDecrementClick}
            type="button"
          >
            <Icon
              source={ChevronDown}
              iconType={IconTypeName.Primary}
              disabled={disabled || hasReachedTheLimit(value, min)}
            />
          </button>
        </>
      )}
    </div>
  );
};
