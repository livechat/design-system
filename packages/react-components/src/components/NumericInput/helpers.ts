export const sumString = (val1: string, val2: string): string => {
  const sum = parseInt(val1, 10) + parseInt(val2, 10);
  return String(sum);
};
