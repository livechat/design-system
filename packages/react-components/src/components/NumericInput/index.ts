export { NumericInput } from './NumericInput';
export { NumericInputField } from './NumericInputField';
