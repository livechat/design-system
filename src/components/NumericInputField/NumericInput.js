import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './style.scss';
import getMergedClassNames from '../../utils/getMergedClassNames';
import { KeyCodes } from '../../constants/keyCodes';

const cx = classNames.bind(styles);
const baseClass = 'numeric-input';

class NumericInput extends React.PureComponent {
  componentDidMount() {
    if (this.props.max && this.props.value > this.props.max) {
      this.props.onChange(this.props.max);
    }

    if (this.props.min && this.props.value < this.props.min) {
      this.props.onChange(this.props.min);
    }
  }

  componentWillUnmount() {
    this.removeKeyboardEventListeners();
  }

  onFocus = () => {
    this.addKeyboardEventListeners();
  };

  onBlur = () => {
    this.removeKeyboardEventListeners();
  };

  onKeyDown = e => {
    if (e.keyCode === KeyCodes.arrowDown) {
      e.preventDefault();
      this.changeValue(-1)();
    }
    if (e.keyCode === KeyCodes.arrowUp) {
      e.preventDefault();
      this.changeValue(1)();
    }
  };

  addKeyboardEventListeners = () => {
    document.addEventListener('keydown', this.onKeyDown);
  };

  removeKeyboardEventListeners = () => {
    document.removeEventListener('keydown', this.onKeyDown);
  };

  changeValue = val => () => {
    if (this.props.value !== null) {
      this.props.onChange(this.calcValue(parseInt(this.props.value, 10) + val));
    } else {
      this.props.onChange(val);
    }
  };

  handleChange = e => {
    e.preventDefault();
    e.stopPropagation();
    const { onChange, value } = this.props;
    const inputVal = e.target.value.replace(/!^-?\d+/, '');
    const val = parseInt(inputVal, 10);

    if (inputVal === '') {
      onChange(null);
      return;
    }

    if (String(val) !== inputVal) {
      onChange(value);
    } else {
      onChange(this.calcValue(val));
    }
  };

  calcValue(val) {
    const { max, min } = this.props;

    if (max && val > max) {
      return max;
    }

    if (min && val < min) {
      return min;
    }

    return val;
  }

  render() {
    const {
      error,
      className,
      max,
      min,
      onChange,
      value,
      disabled,
      style,
      ...restProps
    } = this.props;

    const mergedClassNames = getMergedClassNames(
      cx({
        [baseClass]: true,
        [`${baseClass}--error`]: error,
        [`${baseClass}--disabled`]: disabled
      }),
      className
    );

    return (
      <span className={mergedClassNames} style={style}>
        <input
          type="text"
          {...restProps}
          value={value !== null ? value : ''}
          disabled={disabled}
          onChange={this.handleChange}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        />
        <span
          aria-label="Increment value"
          className={cx({
            [`${baseClass}__increment`]: true,
            [`${baseClass}__increment--disabled`]:
              this.props.max && this.props.value === this.props.max
          })}
          onClick={this.changeValue(1)}
        />
        <span
          aria-label="Decrement value"
          className={cx({
            [`${baseClass}__decrement`]: true,
            [`${baseClass}__decrement--disabled`]:
              this.props.min && this.props.value === this.props.min
          })}
          onClick={this.changeValue(-1)}
        />
      </span>
    );
  }
}

NumericInput.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  error: PropTypes.string,
  value: PropTypes.number,
  max: PropTypes.number,
  min: PropTypes.number,
  disabled: PropTypes.bool,
  onChange: PropTypes.func.isRequired
};

export default NumericInput;
