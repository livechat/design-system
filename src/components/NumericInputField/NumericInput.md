<h3>Numeric Input</h3>

Use when the input requires a numerical value.

```js
initialState = { value: '1', error: null };

requiredValidator = value => value !== '' && value !== '-';
handleInputChange = (value) => {
  setState({
    value,
    error: requiredValidator(value) ? null : 'Field required'
  });
}

<NumericInput
  width="60px"
  min={1}
  value={state.value}
  error={state.error}
  onChange={handleInputChange}
/>
```
```js noeditor
<ComponentHtmlMarkup>
  <NumericInput
    width="60px"
    value="1"
    error="Field required"
    onChange={() => {}}
  />
</ComponentHtmlMarkup>
```

<h3>Numeric Input Field</h3>

You can pass min and max prop to component. In this example value should be within range [-10, 1000]

```js
initialState = { value: '1000', error: null };

requiredValidator = value => value !== '' && value !== '-';
handleInputChange = (value) => {
  setState({
    value,
    error: requiredValidator(value) ? null : 'Field required'
  });
}

<NumericInputField
  id="num-input-field"
  min={-10}
  max={1000}
  width="75px"
  value={state.value}
  error={state.error}
  onChange={handleInputChange}
  labelText="Number"
  description="Choose number"
/>
```
```js noeditor
<ComponentHtmlMarkup>
  <NumericInput
    id="num-input-field"
    min={-10}
    max={1000}
    width="75px"
    value="1000"
    error="Field required"
    onChange={() => {}}
    labelText="Number"
    description="Choose number"
  />
</ComponentHtmlMarkup>
```

<h3>Numeric Input without controls</h3>

```js
initialState = { value: '51' };

handleInputChange = (value) => {
  setState({
    value
  });
}

<NumericInput
  width="35px"
  value={state.value}
  noControls
  onChange={handleInputChange}
/>
```
```js noeditor
<ComponentHtmlMarkup>
  <NumericInput
    width="35px"
    value="51"
    noControls
    onChange={() => {}}
  />
</ComponentHtmlMarkup>
```

<h3>Disabled Numeric Input</h3>

```js
initialState = { value: '51' };
requiredValidator = value => value !== '' && value !== '-';

handleInputChange = (value) => {
  setState({
    value,
  });
}

<NumericInput
  min={1}
  max={50}
  disabled
  value="1"
  onChange={() => {}}
/>
```
```js noeditor
<ComponentHtmlMarkup>
  <NumericInput
    min={1}
    max={50}
    disabled
    value="1"
    onChange={() => {}}
  />
</ComponentHtmlMarkup>
```
