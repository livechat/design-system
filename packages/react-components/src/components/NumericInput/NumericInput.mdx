import { Meta, Title, Canvas, ArgTypes } from '@storybook/addon-docs/blocks';

import * as NumericInput from './NumericInput.stories';

<Meta of={NumericInput} />

<Title>NumericInput</Title>

[Intro](#Intro) | [Component API](#ComponentAPI) | [Content Spec](#ContentSpec)

## Intro <a id="Intro" />

<Canvas of={NumericInput.NumericInput} sourceState="none" />

#### Example implementation

```jsx
<NumericInput value={value} onChange={(v) => setValue(v)} />
```

## Component API <a id="ComponentAPI" />

<ArgTypes of={NumericInput.NumericInput} sort="requiredFirst" />

## Content Spec <a id="ContentSpec" />

<a
  className="sb-unstyled"
  href="https://www.figma.com/file/9FDwjR8lYvincseDkKypC4/Component-Documentations?type=design&node-id=775-65281&mode=dev"
  target="_blank"
>
  Go to Figma documentation
</a>
