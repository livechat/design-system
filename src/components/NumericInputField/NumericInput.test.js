import * as React from 'react';
import { shallow } from 'enzyme';
import NumericInput from './NumericInput';

const generateMockChangeEvent = value => ({
  preventDefault: () => {},
  stopPropagation: () => {},
  target: {
    value
  }
});

describe('NumericInput', () => {
  it('should render NumericInput with required props', () => {
    const changeFn = jest.fn();
    const component = shallow(<NumericInput value="1" onChange={changeFn} />);

    expect(component).toMatchSnapshot();
  });

  it('should render NumericInput with all available props', () => {
    const changeFn = jest.fn();
    const component = shallow(
      <NumericInput
        className="numeric-test"
        value=""
        min={1}
        max={100}
        error="Field required"
        onChange={changeFn}
        disabled
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should call change function with calculated value on input change', () => {
    const changeFn = jest.fn();
    const component = shallow(<NumericInput value="1" onChange={changeFn} />);
    const inputEl = component.find('input');

    inputEl.simulate('change', generateMockChangeEvent(''));
    expect(changeFn).toHaveBeenCalledWith('');

    inputEl.simulate('change', generateMockChangeEvent('3'));
    expect(changeFn).toHaveBeenCalledWith('3');
  });

  it('should call change function with calculated value on inc/dec button click', () => {
    const changeFn = jest.fn();
    const component = shallow(<NumericInput value="1" onChange={changeFn} />);

    const incrementBtn = component.find('[aria-label="Increment value"]');
    const decrementBtn = component.find('[aria-label="Decrement value"]');

    incrementBtn.simulate('click');
    expect(changeFn).toHaveBeenCalledWith('2');

    decrementBtn.simulate('click');
    expect(changeFn).toHaveBeenCalledWith('0');
  });

  it('should not allow values smaller than passed min prop', () => {
    const changeFn = jest.fn();
    const component = shallow(
      <NumericInput min={50} value="50" onChange={changeFn} />
    );

    const inputEl = component.find('input');
    const decrementBtn = component.find('[aria-label="Decrement value"]');

    inputEl.simulate('change', generateMockChangeEvent('49'));
    expect(changeFn).toHaveBeenCalledWith('50');

    decrementBtn.simulate('click');
    expect(changeFn).toHaveBeenCalledWith('50');
  });

  it('should not allow values larger than passed max prop', () => {
    const changeFn = jest.fn();
    const component = shallow(
      <NumericInput max={50} value="50" onChange={changeFn} />
    );

    const inputEl = component.find('input');
    const incrementBtn = component.find('[aria-label="Increment value"]');

    inputEl.simulate('change', generateMockChangeEvent('100'));
    expect(changeFn).toHaveBeenCalledWith('50');

    incrementBtn.simulate('click');
    expect(changeFn).toHaveBeenCalledWith('50');
  });
});
