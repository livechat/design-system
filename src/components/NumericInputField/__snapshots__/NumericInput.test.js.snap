// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput should render NumericInput with all available props 1`] = `
<div
  className="numeric-input numeric-input--error numeric-input--disabled numeric-test"
  style={Object {}}
>
  <input
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
  <button
    aria-label="Increment value"
    className="numeric-input__increment"
    disabled={true}
    onClick={[Function]}
    tabIndex="-1"
  />
  <button
    aria-label="Decrement value"
    className="numeric-input__decrement"
    disabled={true}
    onClick={[Function]}
    tabIndex="-1"
  />
</div>
`;

exports[`NumericInput should render NumericInput with required props 1`] = `
<div
  className="numeric-input"
  style={Object {}}
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value="1"
  />
  <button
    aria-label="Increment value"
    className="numeric-input__increment"
    onClick={[Function]}
    tabIndex="-1"
  />
  <button
    aria-label="Decrement value"
    className="numeric-input__decrement"
    onClick={[Function]}
    tabIndex="-1"
  />
</div>
`;
